### DISCUSSION TITLE
change filetype of sd-image

### DISCUSSION THREAD

**Discussion by Strix-Vyxlor** [nix-community/raspberry-pi-nix#72] [Q&A]
Hello everybody,

I was wondering if I could change the filesystem type of the sd-image.
I would like to use btrfs instead of ext4.

I have taken a look in the source code, but the filesystem type is locked.
Do I just override this in my code, and how.

**Comment by tstat** [comment]
I was just thinking about a pi I would like to use btrfs on yesterday. I don't see a way to override this right now, but I do intend to work on this.

**Comment by tstat** [comment]
I broke off the hardcoded sd-image stuff into a separate module in https://github.com/nix-community/raspberry-pi-nix/pull/79. Using this branch, and following the standard nixos-install process, I have a pi 5 with btrfs on gpt. So, I think making the next steps would be to make a guide / figure out what can be made easier.

Building images formatted this way directly would also be nice.
